### ISSUE TITLE
"Parsing Error Unexpected Token" when destructuring object without using var 

### ISSUE THREAD

**Issue by aneilbaboo** [eslint/eslint#7832]
**Tell us about your environment**

* **ESLint Version:** v3.11.1
* **Node Version:** v7.2.0
* **npm Version:** 3.10.9

**What parser (default, Babel-ESLint, etc.) are you using?**
Mainly es-lint, but also tried babel-eslint 6.12

**Please show your full configuration:**
```yaml
env:
  es6: true
  node: true
  mocha: true
extends: 'eslint:recommended'
parserOptions:
  sourceType: module
  ecmaFeatures:
        experimentalObjectRestSpread: true
plugins:
  - mocha
rules:
  "strict": 0
  "no-unused-vars":
    - error
    - vars: all
       varsIgnorePattern: "^_$"
  indent:
    - error
    - 2
  linebreak-style:
    - error
    - unix
  quotes:
    - off
    - double
  semi:
    - error
    - always
  "mocha/no-exclusive-tests": error
```
**What did you do? Please include the actual source code causing the issue.**

```js
var a,b;

{a,b} = {a:1, b:2}; // parsing error at "= {a: 1"
```

**What did you expect to happen?**

This seems to be valid even in node v7.2.0 without babel.

**What actually happened? Please include the actual, raw output from ESLint.**
```shell
$ eslint sandbox/sandbox.js 

/Users/aneil/sandbox/sandbox.js
  3:8  error  Parsing error: Unexpected token

✖ 1 problem (1 error, 0 warnings)
```

**Comment by aneilbaboo** [comment]
Closing.  I fixed it by getting rid of an error in my config (missing "-" in front of varsIgnorePattern).

**Comment by aneilbaboo** [comment]
Actually, this is still valid.  I fixed the error, and I'm still getting the behavior.  I amended my config above.

**Comment by platinumazure** [comment]
Hi @aneilbaboo, thanks for the issue. 

You're getting a parsing error because, without a declaration, object destructure assignment looks like it could be a block on the left side and the ES2015 spec says it must be parsed as a block. 

You can fix this by surrounding with parentheses so the engine knows it's an expression: 

```js
({a,b} = {a:1, b:2});
```

See [here](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Destructuring_assignment#Assignment_without_declaration) for more information.

**Comment by platinumazure** [comment]
Also, just a tip: ESLint has to successfully parse a file before *any* lint rules can be run on that file. So if you're getting a "Parser error", changing your rule configuration generally won't help. It's best to try to fix the parser error first.

Hope this helps you in future :smile:

**Comment by aneilbaboo** [comment]
@platinumazure - Ah, thank you very much!  I was getting tripped up because node 7.2.0 accepts this as valid syntax.  Closing.
